micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -178,74 +178,76 @@ void checkAndBindMetrics(MeterRegistry registry) {
+                // Filter out non-numeric values
+                // Filter out metrics from groups that include metadata
+                if (!(metric.metricValue() instanceof Number) || METRIC_GROUP_APP_INFO.equals(name.group())
+                        || METRIC_GROUP_METRICS_COUNT.equals(name.group())) {
+                    return;
+                }
 
-                    String meterName = meterName(name);
-
-                    // Kafka has metrics with lower number of tags (e.g. with/without
-                    // topic or partition tag)
-                    // Remove meters with lower number of tags
-                    boolean hasLessTags = false;
-                    for (Meter other : registryMetersByNames.getOrDefault(meterName, emptyList())) {
-                        Meter.Id otherId = other.getId();
-                        List<Tag> tags = otherId.getTags();
-                        List<Tag> meterTagsWithCommonTags = meterTags(name, true);
-                        if (tags.size() < meterTagsWithCommonTags.size()) {
-                            registry.remove(otherId);
-                            registeredMeterIds.remove(otherId);
-                        }
-                        // Check if already exists
-                        else if (tags.size() == meterTagsWithCommonTags.size()) {
-                            if (tags.containsAll(meterTagsWithCommonTags))
-                                return;
-                        }
-                        else
-                            hasLessTags = true;
+                String meterName = meterName(name);
+
+                // Kafka has metrics with lower number of tags (e.g. with/without
+                // topic or partition tag)
+                // Remove meters with lower number of tags
+                boolean hasLessTags = false;
+                for (Meter other : registryMetersByNames.getOrDefault(meterName, emptyList())) {
+                    Meter.Id otherId = other.getId();
+                    List<Tag> otherTags = otherId.getTags();
+                    List<Tag> meterTagsWithCommonTags = meterTags(name, true);
+                    if (otherTags.size() < meterTagsWithCommonTags.size()) {
+                        registry.remove(otherId);
+                        registeredMeterIds.remove(otherId);
                     }
-                    if (hasLessTags)
-                        return;
-
-                    List<Tag> tags = meterTags(name);
-                    try {
-                        Meter meter = bindMeter(registry, metric, meterName, tags);
-                        List<Meter> meters = registryMetersByNames.computeIfAbsent(meterName, k -> new ArrayList<>());
-                        meters.add(meter);
+                    // Check if already exists
+                    else if (otherTags.size() == meterTagsWithCommonTags.size()) {
+                        if (new HashSet<>(otherTags).containsAll(meterTagsWithCommonTags))
```

Comment:
This change was suggested by IntelliJ. See https://www.jetbrains.com/help/inspectopedia/SlowListContainsAll.html

Reply:
Could you please add this as a comment so the next person who looks at this will know the reason?

Reply:
Done